article/app/controllers/ArticleController.scala

```diff
@@ -180,128 +34,61 @@ class ArticleController(contentApiClient: ContentApiClient, val controllerCompon
-    lookup(path, range) map responseToModelOrResult(range, asArticle) recover convertApiExceptions map {
-      case Left(model) => render(model)
-      case Right(other) => RenderOtherStatus(other)
+  def renderEmail(path: String): Action[AnyContent] = {
+    Action.async { implicit request =>
+      mapModel(path, range = Some(ArticleBlocks)) {
+        render(path, _)
+      }
     }
   }
 
-  private def lookup(path: String, range: Option[BlockRange])(implicit request: RequestHeader): Future[ItemResponse] = {
-    val edition = Edition(request)
-
-    val capiItem = contentApiClient.item(path, edition)
-      .showTags("all")
-      .showFields("all")
-      .showReferences("all")
-      .showAtoms("all")
-
-    val capiItemWithBlocks = range.map { blockRange =>
-      val blocksParam = blockRange.query.map(_.mkString(",")).getOrElse("body")
-      capiItem.showBlocks(blocksParam)
-    }.getOrElse(capiItem)
-    contentApiClient.getResponse(capiItemWithBlocks)
+  private def render(path: String, page: PageWithStoryPackage)(implicit request: RequestHeader): Future[Result] = page match {
+    case article: ArticlePage =>
+      val htmlResponse = () => {
+        if (request.isEmail) ArticleEmailHtmlPage.html(article)
+        else if (request.isAmp) views.html.articleAMP(article)
+        else ArticleHtmlPage.html(article)
+      }
+      // add extra data for *.json?guui endpoint
+      val contentFieldsJson = if (request.isGuuiJson) List("contentFields" -> Json.toJson(ContentFields(article.article))) else List()
+      val jsonResponse = () => List(("html", views.html.fragments.articleBody(article))) ++ contentFieldsJson
+      Future { renderFormat(htmlResponse, jsonResponse, article) }
+  }
 
+  def mapModel(path: String, range: Option[BlockRange] = None)(render: PageWithStoryPackage => Future[Result])(implicit request: RequestHeader): Future[Result] = {
+    lookerUpper.lookup(path, range).map((itemResp: ItemResponse) => responseToModelOrResult(range)(itemResp)).recover(convertApiExceptions).flatMap {
+      case Left(model) => render(model)
+      case Right(other) => Future.successful(RenderOtherStatus(other))
+    }
   }
 
-  /**
-   * convert a response into something we can render, and return it
-   * optionally, throw a response if we know it's not right to send the content
-    *
-    * @param response
-   * @return Either[PageWithStoryPackage, Result]
-   */
-  def responseToModelOrResult(range: Option[BlockRange], asArticle:Boolean)(response: ItemResponse)(implicit request: RequestHeader): Either[PageWithStoryPackage, Result] = {
+  def responseToModelOrResult(range: Option[BlockRange])(response: ItemResponse)(implicit request: RequestHeader): Either[PageWithStoryPackage, Result] = {
 
     val supportedContent = response.content.filter(isSupported).map(Content(_))
```

Comment:
isSupported could change to return an Option of an Article, then you could simplify this logic